hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/SparkJob.groovy

```diff
@@ -0,0 +1,182 @@
+    appParams = new LinkedList<String>()
+    setJobProperty("type", "spark");
+  }
+
+  /**
+   * Clones the job.
+   *
+   * @return The cloned job
+   */
+  @Override
+  SparkJob clone() {
+    return clone(new SparkJob(name));
+  }
+
+  /**
+   * Helper method to set the properties on a cloned job.
+   *
+   * @param cloneJob The job being cloned
+   * @return The cloned job
+   */
+  SparkJob clone(SparkJob cloneJob) {
+    cloneJob.properties.putAll(properties);
+    cloneJob.confs.putAll(confs);
+    cloneJob.flags.addAll(flags);
+    cloneJob.appParams.addAll(appParams);
+    cloneJob.appClass = appClass;
+    cloneJob.executionJar = executionJar;
+    return super.clone(cloneJob);
+  }
+
+  /**
+   * DSL method to specify job properties to set on the job. Specifying job properties causes lines
+   * of the form key=val to be written to the job file.
+   * <p>
+   * Additionally for JavaProcessJobs, you can specify JVM properties by using the syntax
+   * "set jvmProperties: [ ... ]", which causes a line of the form
+   * jvm.args=-Dkey1=val1 ... -DkeyN=valN to be written to the job file.
+   * <p>
+   * Additionally for HadoopJavaProcessJob subclasses, you can specify Hadoop job configuration
+   * properties by using the syntax "set confProperties: [ ... ]", which causes lines of the form
+   * hadoop-inject.key=val to be written to the job file.
+   * <p>
+   * Additionally for SparkJobs, you can specify Spark configuration by using the syntax
+   * "set conf: [ ... ]". For each parameter you set, a line of the form param.key=val will
+   * be written to the job file.
+   *
+   * @param args Args whose key 'properties' has a map value specifying the job properties to set;
+   *   or a key 'jvmProperties' with a map value that specifies the JVM properties to set;
+   *   or a key 'confProperties' with a map value that specifies the Hadoop job configuration properties to set;
+   *   or a key 'parameters' with a map value that specifies the Spark parameters to set
+   */
+  @Override
+  void set(Map args) {
+    super.set(args);
+    if (args.containsKey("confs")) {
+      Map<String, Object> confs = args.confs;
+      confs.each() { String name, Object value ->
+        setConf(name, value);
+      }
+    }
+  }
+
+  /**
+   * DSL setConf method sets Spark configurations for the job. When the job file is built, job
+   * properties of the form conf.name=value are added to the job file.
+   *
+   * @param name The Spark parameter name
+   * @param value The Spark parameter value
+   */
+  void setConf(String name, Object value) {
+    confs.put(name, value);
+    setJobProperty("conf.${name}", value);
+  }
+
+  /**
+   * DSL enableFlags method sets the flags for the job. When the job file is build, job properties
+   * of the form flag.flagName=true is added in the job file.
+   * @param flags
+   */
+  void enableFlags(List<String> flags) {
+    this.flags = flags;
+    flags.each { flag ->
+      setJobProperty("flag.$flag",'true');
```

Comment:
Hi @johnyu0520,
I think having ```flag.<flag>=enable/disable``` would be easier to understand than having ```flag.<flag>="doesn't matter"``` in the Azkaban Spark jobtype.

If flag is enabled then we set it else ignore it(default behaviour).